.github/actions/deploy-aio/action.yml

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+# This workflow assumes your cluster is already connected to ARC
+name: 'Deploy Azure IoT Operations to a connected cluster'
+description: |
+  Action to deploy Azure IoT Operations resources to an existing ARC-connected cluster using the `azure-iot-ops` CLI extension.
+  If `config` is provided, it will be written as a file to ~/.kube/config and used to connect to your cluster. 
+  Otherwise, your local `~/.kube/config` file will be used.
+inputs:
+  config:
+    description: |
+      Contents of kubeconfig to connect to cluster. If not provided, we will attempt 
+      to use your local `~/.kube/config` file
+  cluster:
+    description: 'Cluster Name'
+    required: true
+  resource-group:
+    description: 'Resource Group'
+    required: true
+  cluster-namespace:
+    description: The cluster namespace AIO infra will be deployed to.
+  custom-location:
+    description: The custom location namespace corresponding to the AIO deployment
+  disable-rsync-rules:
+    description: Resource sync rules will not be included in the deployment.
+  location:
+    description: The ARM location for RPSaaS collateral.
+  no-deploy:
+    description: The IoT Operations deployment workflow will be skipped.
+  no-preflight:
+    description: Disable preflight checks for deployment
+  no-tls:
+    description: Disable preflight checks for deployment
+  keyvault-id:
+    description: |
+      Keyvault ID to use for AIO deployment SPC.
+      Not required if the AKV CSI driver is already configured on this cluster.
+  mq-insecure:
+    description: Deploy with `--mq-insecure`
+  ca-valid-days:
+    description: Days for self-signed cert to be valid
+  ca-file:
+    description: CA Cert PEM path
```

Comment:
I think we should explicitly include verbiage around "user defined" to contrast with autogenerated test only collateral.